lib/spack/spack/cmd/install.py

```diff
@@ -489,43 +341,164 @@ def install(parser, args):
     if args.deprecated:
         spack.config.set("config:deprecated", True, scope="command_line")
 
-    spack.cmd.common.arguments.sanitize_reporter_options(args)
+    arguments.sanitize_reporter_options(args)
 
     def reporter_factory(specs):
         if args.log_format is None:
-            return None
+            return lang.nullcontext()
 
-        context_manager = spack.report.build_context_manager(
+        return spack.report.build_context_manager(
             reporter=args.reporter(), filename=report_filename(args, specs=specs), specs=specs
         )
-        return context_manager
 
     install_kwargs = install_kwargs_from_args(args)
 
-    if not args.spec and not args.specfiles:
-        # If there are no args but an active environment then install the packages from it.
-        install_all_specs_from_active_environment(
-            install_kwargs=install_kwargs,
-            only_concrete=args.only_concrete,
-            cli_test_arg=args.test,
-            reporter_factory=reporter_factory,
-        )
+    env = ev.active_environment()
+
+    if not env and not args.spec and not args.specfiles:
+        _die_require_env()
+
+    try:
+        if env:
+            install_with_active_env(env, args, install_kwargs, reporter_factory)
+        else:
+            install_without_active_env(args, install_kwargs, reporter_factory)
+    except spack.build_environment.InstallError as e:
+        if args.show_log_on_error:
+            _dump_log_on_error(e)
+        raise
+
+
+# Install inside environments
```

Comment:
Nitpick, but I think these comments out of functions will become obsolete in time. Can we remove them?